src/backend/distributed/connection/connection_management.c

```diff
@@ -1237,7 +1237,8 @@ ConnectionHashHash(const void *key, Size keysize)
 	hash = hash_combine(hash, hash_uint32(entry->port));
 	hash = hash_combine(hash, string_hash(entry->user, NAMEDATALEN));
 	hash = hash_combine(hash, string_hash(entry->database, NAMEDATALEN));
-	hash = hash_combine(hash, hash_uint32(entry->replicationConnParam));
+	hash = hash_combine(hash, hash_any((void *) &entry->replicationConnParam,
+									   sizeof(entry->replicationConnParam)));
```

Comment:
What's a bit weird is that hash_uint32 and hash_any both return Datum whereas other functions return uint32.